lms/djangoapps/grades/tests/test_tasks.py

```diff
@@ -197,7 +197,7 @@ def test_subsection_grades_not_enabled_on_course(self, default_store):
             self.set_up_course(enable_subsection_grades=False)
             self.assertFalse(PersistentGradesEnabledFlag.feature_enabled(self.course.id))
             with check_mongo_calls(0):
-                with self.assertNumQueries(0):
+                with self.assertNumQueries(1):
```

Comment:
This feature works entirely at the level of celery tasks.  It (appropriately) has no knowledge of grades or feature flags.  

If a PersistOnFailureTask is run and it fails, it will make a database query to add the task to the FailedTask table.  

If it runs and it succeeds, it will make a database query to ensure that the named task is removed from the table.  If the task isn't in the table, this query will have no effect, but it still needs to be made to figure out if it needs to do anything.  

I tried a different architecture where we pass in whether the task is a reapplication when we call it, but celery makes it very difficult to add such customizations.  The one way I could see to do it broke testing.

In order to avoid this overhead, we would need to alter the feature flag to prevent calling the task to begin with.  We may already be doing this (I haven't looked), but the test in question calls the task directly, so there's no way to avoid it here.